pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/leap/LeapDriver.java

```diff
@@ -95,33 +93,41 @@ public String initialState() {
         bucket.asScheduler().consume(1);
         final List<PersistentQueueElement> events = new ArrayList<>();
         final HttpClient client = HttpClient.newHttpClient();
-        final String uri = apiUri + "/ClientApi/V1/DeviceReadings";
-        final HttpRequest request = HttpRequest.newBuilder()
-            .uri(URI.create(uri))
-            .header("Authorization", "Bearer " + getAuthToken())
-            .build();
+        final String uri;
+        if (state == "")
+            uri = apiUri + "/ClientApi/V1/DeviceReadings";
+        else {
+            state = dateFormat.format(Date.from(dateFormat.parse(state).toInstant().plus(Duration.ofMillis(1))));
+            uri = apiUri + "/ClientApi/V1/DeviceReadings?startDate=" + state;
+        }
+        final HttpRequest request = HttpRequest.newBuilder().uri(URI.create(uri))
+                .header("Authorization", "Bearer " + getAuthToken()).build();
         log.info("pollEvents: request={}", request);
         final HttpResponse<String> response = client.send(request, BodyHandlers.ofString());
         log.info("pollEvents: response={}", response);
         if (response.statusCode() != 200) {
-            throw new RuntimeException(MessageFormat.format("HTTP server returned status code {0}", response.statusCode()));
-        };
-        log.trace("pollEvents: body={}", response.body());        
-        JsonNode jsonNode = mapper.readTree(response.body());        
+            throw new RuntimeException(
+                    MessageFormat.format("HTTP server returned status code {0}", response.statusCode()));
+        }
+        ;
+        log.trace("pollEvents: body={}", response.body());
+        JsonNode jsonNode = mapper.readTree(response.body());
         final ArrayNode arrayNode = (ArrayNode) jsonNode;
-        Date maxTime = mapper.convertValue(arrayNode.get(0).get("receivedTimestamp"), Date.class);
-
-        final long timestampNanos = System.currentTimeMillis() * 1000 * 1000;
-        for (JsonNode node : arrayNode) {            
-            final byte[] bytes = node.toString().getBytes(StandardCharsets.UTF_8);
-            final String routingKey = getRoutingKey();
-            final PersistentQueueElement event = new PersistentQueueElement(bytes, routingKey, timestampNanos);
-            events.add(event);
-            Date curReading = mapper.convertValue(node.get("receivedTimestamp"),Date.class);
-            if(maxTime.getTime() < curReading.getTime())
-                maxTime = curReading;
+        // if no response, empty event list and same state will be returned
+        if (arrayNode != null) {
```

Comment:
Shouldn't this also check for `arrayNode.size() == 0`?

Reply:
Added the check